src/structures/BaseGuildVoiceChannel.js

```diff
@@ -68,8 +68,13 @@ class BaseGuildVoiceChannel extends GuildChannel {
    */
   get joinable() {
     if (!this.viewable) return false;
-    if (!this.permissionsFor(this.client.user).has(Permissions.FLAGS.CONNECT, false)) return false;
-    return true;
+    const permissions = this.permissionsFor(this.client.user);
+    if (!permissions) return false;
+
+    // This flag allows joining even if timed out
+    if (permissions.has(Permissions.FLAGS.ADMINISTRATOR, false)) return true;
+    if (!permissions.has(Permissions.FLAGS.CONNECT, false)) return false;
+    return this.guild.me.communicationDisabledUntilTimestamp < Date.now();
```

Comment:
You could do an `ADMINISTRATOR` check and then `return` if the you (don't) have `timeout` and have `CONNECT`
```suggestion
    return this.guild.me.communicationDisabledUntilTimestamp < Date.now() && permissions.has(Permissions.FLAGS.CONNECT, false);
```